File: app/src/main/java/co/early/password123/feature/analytics/Analytics.java
package co.early.password123.feature.analytics;

import android.support.annotation.Nullable;

import com.crashlytics.android.answers.Answers;
import com.crashlytics.android.answers.CustomEvent;

import co.early.asaf.core.Affirm;
import co.early.asaf.core.logging.Logger;


public class Analytics {

    private static final String TAG_ONLINE = "Antics:ONLINE";
    private static final String TAG_OFFLINE = "Antics:OFFLINE";

    @Nullable
    private final Answers answers;
    private final Logger logger;

    public Analytics(@Nullable Answers answers, Logger logger) {
        this.answers = answers;
        this.logger = Affirm.notNull(logger);
    }


    public void logEventRemotelyOrLocally(String evetName){

        Affirm.notNull(evetName);

        if (answers != null){
            answers.logCustom(new CustomEvent(evetName));
            logger.i(TAG_ONLINE, evetName);
        }else{
            logger.i(TAG_OFFLINE, evetName);
        }
    }

}

File: app/src/main/java/co/early/password123/feature/analytics/Analytics.kt
package co.early.password123.feature.analytics

import co.early.asaf.core.Affirm
import co.early.asaf.core.logging.Logger
import com.crashlytics.android.answers.Answers
import com.crashlytics.android.answers.CustomEvent


class Analytics(private val answers: Answers?, logger: Logger) {

    private val logger: Logger

    init {
        this.logger = Affirm.notNull(logger)
    }


    fun logEventRemotelyOrLocally(eventName: String) {

        Affirm.notNull(eventName)

        if (answers != null) {
            answers.logCustom(CustomEvent(eventName))
            logger.i(TAG_ONLINE, eventName)
        } else {
            logger.i(TAG_OFFLINE, eventName)
        }
    }

    companion object {

        private val TAG_ONLINE = "Analytics:ON"
        private val TAG_OFFLINE = "Analytics:OFF"
    }

}
